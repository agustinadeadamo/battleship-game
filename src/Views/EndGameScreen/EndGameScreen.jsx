/**
 * @desc Dependencies
 */
import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';

/**
 * @desc styled components
 */
import EndGameScreen from './style';

/**
 * @desc Actions
 */
import {
  updateUserName,
} from '../../Actions/Actions';

const StartScreen = () => {
  // Store
  const winner = useSelector((state) => state.winner);
  const dispatch = useDispatch();

  // Router
  const history = useHistory();

  /**
   * Function that is executed when play again button is clicked
   */
  const onClickPlayAgainButton = () => {
    // Changes name in storeto empty string and redirects to initial view
    dispatch(updateUserName(''));
    history.push('/');
  };

  return (
    <EndGameScreen data-testid="end-game-container">
      <div className="winner-container">
        <p className="winner">
          The winner is:
        </p>
        <p className="winner">
          {winner}
        </p>
        <button data-testid="play-again-button" onClick={() => onClickPlayAgainButton()} type="button" className="dark button-play-again">Play again</button>
      </div>
    </EndGameScreen>
  );
};

export default StartScreen;
