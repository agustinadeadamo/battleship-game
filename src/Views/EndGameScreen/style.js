/**
 * @desc Dependencies
 */
import styled from 'styled-components';

const EndGameScreen = styled.div`
width: 90%;
max-width: 1240px;
margin: 0 auto;
display: flex;
flex-direction: column;
justify-content: center;
align-items: center;
text-align: center;
height: 100vh;

  .winner-container {
    width: 300px;
    height: 200px;
    

    .winner {
      font-size: 30px;
      text-align: center;
    }

    .button-play-again {
      margin: 30px 0;
    }
  }
`;

export default EndGameScreen;
