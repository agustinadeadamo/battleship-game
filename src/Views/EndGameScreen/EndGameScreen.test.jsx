/**
 * @desc Dependencies
 */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { act, create } from 'react-test-renderer';
import {
  describe, expect, it, beforeEach,
} from '@jest/globals';

/**
 * @desc Store
 */
import Store from '../../Config/Store';

/**
 * @desc Components
 */
import EndGameScreen from './EndGameScreen';

// Mocks react router history
const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('[EndGameScreen View]', () => {
  let getByTestId;

  beforeEach(() => {
    ({ getByTestId } = render(
      <Provider store={Store}>
        <EndGameScreen />
      </Provider>,
    ));
  });

  it('test if renders properly', () => {
    // Gets endGame container
    const EndGameContainer = getByTestId('end-game-container');
    expect(EndGameContainer).toBeInTheDocument();
  });

  it('Matchs snapshot', () => {
    const startscreen = create(
      <Provider store={Store}>
        <EndGameScreen />
      </Provider>,
    );
    expect(startscreen).toMatchSnapshot();
  });

  it('test if onClick play again button generates redirects', () => {
    // Gets play again button
    const playAgainButton = getByTestId('play-again-button');
    // Fires onClick event to play again
    act(() => { fireEvent.click(playAgainButton); });
    expect(mockHistoryPush).toHaveBeenCalledWith('/');
  });
});
