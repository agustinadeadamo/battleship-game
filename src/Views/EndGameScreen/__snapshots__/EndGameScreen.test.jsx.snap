// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[EndGameScreen View] Matchs snapshot 1`] = `
<div
  className="sc-bdfBwQ jDcNjQ"
  data-testid="end-game-container"
>
  <div
    className="winner-container"
  >
    <p
      className="winner"
    >
      The winner is:
    </p>
    <p
      className="winner"
    >
      
    </p>
    <button
      className="dark button-play-again"
      data-testid="play-again-button"
      onClick={[Function]}
      type="button"
    >
      Play again
    </button>
  </div>
</div>
`;
